### PR TITLE
Allow init container with name

### PR DESCRIPTION
**Pull request by kidsbear**
Allow client side to initialize a container with specific container name instead of providing prefix.

### CODE REVIEW

**Comment by kidsbear** [comment]
@ratulm We don't have Option class for client now right? For this particular one, I just compare the option string, should I create a new Option class for it?

**Comment by ratulm** [comment]
what will the Option class do?

**Comment by kidsbear** [comment]
Defines a bunch of constant options? For now we only have one option(-setname), if we are planning to add more options in the future, an Option class will make it easier to control?

**Comment by ratulm** [comment]
i see. good idea but i'd say lets skip that generalization for now; we can get to it later if/when we add more options.

File: projects/batfish-client/src/main/java/org/batfish/client/Command.java [hunk 1]
```diff
@@ -173,7 +173,7 @@
                   "<analysis-name> <question-directory>",
                   "Initialize a new analysis for the container"));
       descs.put(INIT_CONTAINER, new Pair<>(
-            "[<container-name-prefix>]",
+            "<-setname><container-name> or [<container-name-prefix>]",
```
Comment (dhalperi) [inline projects/batfish-client/src/main/java/org/batfish/client/Command.java]: this does not seem like the right syntax. 

`[-setname <container-name>] or [<container-name-prefix>]` seems closer. But probably it should be a multiline string with different options on different lines. Are there existing examples to mimic?

File: projects/batfish-client/src/main/java/org/batfish/client/Client.java [hunk 2]
```diff
@@ -1501,15 +1501,31 @@ private boolean help(List<String> options, List<String> parameters) {
...
-         _logger.errorf("Invalid arguments: %s\n", Arrays.toString(words));
-         printUsage(Command.INIT_CONTAINER);
+   private boolean initContainer(List<String> options, List<String> parameters) {
+      if (!isValidArgument(options, parameters, 1, 0, 1, Command.INIT_CONTAINER)) {
          return false;
       }
-      String containerPrefix = (words.length > 1) ? words[1]
-            : DEFAULT_CONTAINER_PREFIX;
-      _currContainerName = _workHelper.initContainer(null, containerPrefix);
+      if (!options.isEmpty()) {
```
Comment (dhalperi) [inline projects/batfish-client/src/main/java/org/batfish/client/Client.java]: It's okay, but I bet you could simplify this code a lot by ordering your `if`s better and unnesting. The current code is way more complex than it needs to be IMO ; hard to understand.

Example:

```java
if (options.contains("-setname")) {
   validate using =1 option, using =1 parameter
   _currContainerName = _workHelper.initCountainer(parameters.get(0), null));
}
else {
   validate using =0 options, 0 or 1 parameter.
   existing 2 lines of logic
}
final 3 lines of logs.
```

File: projects/coordinator/src/test/java/org/batfish/coordinator/WorkMgrTest.java [hunk 3]
```diff
@@ -87,4 +88,41 @@ public void listSortedQuestionNames() throws IOException {
       assertThat(questions.size(), is(3));
       assertThat(questions.toString(), equalTo("[access, initinfo, nodes]"));
    }
+
+   // Tests for initContainer method
```
Comment (dhalperi) [inline projects/coordinator/src/test/java/org/batfish/coordinator/WorkMgrTest.java]: drop this comment -- such comments are unnecessary as the test name should make it clear what it's testing.

**Comment by dhalperi** [comment]
Nice!

**Pull request merged**